modules/programs/ssh.nix

```diff
@@ -16,63 +16,223 @@ let
 
   cfg = config.programs.ssh;
 
+  mkSshOptions =
```

Comment:
Would probably recommend pulling all the helper functions out into a new `lib.nix` under a new `ssh` folder to keep the module from getting too bloated.